src/3d/qgscameracontroller.cpp

```diff
@@ -379,48 +381,172 @@ void QgsCameraController::onPositionChangedTerrainNavigation( Qt3DInput::QMouseE
 
   if ( ( hasLeftButton && hasShift && !hasCtrl ) || ( hasMiddleButton && !hasShift && !hasCtrl ) )
   {
-    // rotate/tilt using mouse (camera moves as it rotates around its view center)
-    float pitch = mCameraPose.pitchAngle();
-    float yaw = mCameraPose.headingAngle();
-    pitch += 0.2f * dy;
-    yaw -= 0.2f * dx;
-    mCameraPose.setPitchAngle( pitch );
-    mCameraPose.setHeadingAngle( yaw );
-    updateCameraFromPose();
+    // rotate/tilt using mouse (camera moves as it rotates around the clicked point)
+
+    double scale = std::max( mViewport.width(), mViewport.height() );
+    float pitchDiff = 180 * ( mouse->y() - mMiddleButtonClickPos.y() ) / scale;
+    float yawDiff = -180 * ( mouse->x() - mMiddleButtonClickPos.x() ) / scale;
+
+    if ( !mDepthBufferIsReady )
+      return;
+
+    if ( !mRotationCenterCalculated )
+    {
+      double depth = Qgs3DUtils::decodeDepth( mDepthBufferImage.pixelColor( mMiddleButtonClickPos.x(), mMiddleButtonClickPos.y() ) ); //depthPixel.redF() / 255.0 / 255.0 + depthPixel.greenF() / 255.0 + depthPixel.blueF();
```

Comment:
Please remove the commented out code here and in the other places where Qgs3DUtils::decodeDepth() is now being used.